client/finality-grandpa/rpc/src/lib.rs

```diff
@@ -166,6 +296,68 @@ mod tests {
 			\"setId\":1\
 		},\"id\":1}";
 
-		assert_eq!(io.handle_request_sync(request), Some(response.into()));
+		let meta = sc_rpc::Metadata::default();
+		assert_eq!(io.handle_request_sync(request, meta), Some(response.into()));
+	}
+
+	fn setup_pubsub_handler() -> (sc_rpc::Metadata, jsonrpc_pubsub::PubSubHandler<sc_rpc::Metadata>) {
+		let handler = setup_rpc_handler(TestVoterState);
+		let mut io = jsonrpc_pubsub::PubSubHandler::new(jsonrpc_core::MetaIoHandler::default());
```

Comment:
Why do you need `PubSubHandler` here?

Reply:
Changing it to `MetaIoHandler` does indeed still pass the tests. But why isn't `PubSubHandler` needed for pubsub rpc methods?

Reply:
`PubSub` stuff is just a "syntactic sugar" over regular `MetaIoHandler`. If you take a look at the types the `PubSubHandler` adds convenience methods to add subscriptions, but here we don't add methods directly but rather `.extend` the handler with methods generated by the proc macro (the trait stuff) - this is something that can be easily done with `MetaIoHandler` alone.

The only place where pubsub is required is the `Metadata` bound - the `Metadata` needs to support providing session object, but since we use concrete `sc_rpc::Metadata` everywhere it does satisfy that bound.